src/NuGet.Core/NuGet.Commands/TrustedSignersCommand/TrustedSignerActionsProvider.cs

```diff
@@ -0,0 +1,304 @@
+        public async Task SyncTrustedRepositoryAsync(string name, CancellationToken token)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentException(Strings.ArgumentCannotBeNullOrEmpty, nameof(name));
+            }
+
+            var signers = _trustedSignersProvider.GetTrustedSigners();
+            foreach (var existingRepository in signers.OfType<RepositoryItem>())
+            {
+                if (string.Equals(existingRepository.Name, name, StringComparison.Ordinal))
+                {
+                    var certificateItems = await GetCertificateItemsFromServiceIndexAsync(existingRepository.ServiceIndex, token);
+
+                    existingRepository.Certificates.Clear();
+                    existingRepository.Certificates.AddRange(certificateItems);
+
+                    _trustedSignersProvider.AddOrUpdateTrustedSigner(existingRepository);
+
+                    return;
+                }
+            }
+
+            throw new InvalidOperationException(string.Format(CultureInfo.CurrentCulture, Strings.Error_TrustedRepositoryDoesNotExist, name));
+        }
+
+#if IS_DESKTOP
+        /// <summary>
+        /// Adds a trusted signer item to the settings based a signed package.
+        /// </summary>
+        /// <param name="name">Name of the trusted signer.</param>
+        /// <param name="package">Package to read signature from.</param>
+        /// <param name="trustTarget">Signature to trust from package.</param>
+        /// <param name="allowUntrustedRoot">Specifies if allowUntrustedRoot should be set to true.</param>
+        /// <param name="owners">Trusted owners that should be set when trusting a repository.</param>
+        /// <param name="token">Cancellation token for async request</param>
+        public async Task AddTrustedSignerAsync(string name, ISignedPackageReader package, VerificationTarget trustTarget, bool allowUntrustedRoot, IEnumerable<string> owners, CancellationToken token)
+        {
+            if (package == null)
+            {
+                throw new ArgumentNullException(nameof(package));
+            }
+
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentException(Strings.ArgumentCannotBeNullOrEmpty, nameof(name));
+            }
+
+            if (!Enum.IsDefined(typeof(VerificationTarget), trustTarget) || (trustTarget != VerificationTarget.Repository && trustTarget != VerificationTarget.Author))
+            {
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.Error_UnsupportedTrustTarget, trustTarget.ToString()));
+            }
+
+            if (trustTarget == VerificationTarget.Author && owners != null && owners.Any())
+            {
+                throw new ArgumentException(Strings.Error_TrustedAuthorNoOwners);
+            }
+
+            var v3ServiceIndex = string.Empty;
```

Comment:
`string v3ServiceIndex = null;`

Don't use `string.Empty` as a default.  `null` represents the absence of a value, which is more appropriate than `string.Empty`.